src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2007,5 +2008,42 @@ private IOperation CreateUsingLocalDeclarationsOperation(BoundUsingLocalDeclarat
                                                  getChildren: () => ImmutableArray.Create<IOperation>(CreateBoundMultipleLocalDeclarationsOperation((BoundMultipleLocalDeclarations)boundNode)),
                                                  isImplicit: false);
         }
+
+        internal IPropertySubpatternOperation CreatePropertySubpattern(BoundSubpattern subpattern)
+        {
+            SyntaxNode syntax = subpattern.Syntax;
+            return new CSharpLazyPropertySubpatternOperation(this, subpattern, syntax, _semanticModel);
+        }
+
+        internal IOperation CreatePropertySubpatternMember(Symbol symbol, SyntaxNode syntax)
+        {
+            var nameSyntax = (syntax is SubpatternSyntax subpatSyntax ? subpatSyntax.NameColon?.Name : null) ?? syntax;
+            bool isImplicit = nameSyntax != syntax;
+            switch (symbol)
+            {
+                case FieldSymbol field:
+                    {
+                        var constantValue = field.ConstantValue is null ? default(Optional<object>) : new Optional<object>(field.ConstantValue);
+                        var receiver = new InstanceReferenceOperation(
+                            InstanceReferenceKind.PatternInput, _semanticModel, nameSyntax, field.ContainingType, constantValue, isImplicit: true);
+                        return new FieldReferenceOperation(
+                            field, isDeclaration: false, receiver, _semanticModel, nameSyntax, field.Type.TypeSymbol, constantValue, isImplicit: isImplicit);
+                    }
+                case PropertySymbol property:
+                    {
+                        var receiver = new InstanceReferenceOperation(
+                            InstanceReferenceKind.PatternInput, _semanticModel, nameSyntax, property.ContainingType, constantValue: default, isImplicit: true);
+                        return new PropertyReferenceOperation(
+                            property, receiver, ImmutableArray<IArgumentOperation>.Empty, _semanticModel, nameSyntax, property.Type.TypeSymbol,
```

Comment:
>ImmutableArray<IArgumentOperation>.Empty [](start = 48, length = 40)

In an error case, could we ran into a parameterized property?

Reply:
It does not appear so. The closest I could get was an `IConstantPatternOperation` child of the `IPropertyPatternOperation`. Using an indexer like below doesn't parse close to cleanly at all, it results in multiple `IncompleteMember`s on the class.
```C#
class C
{
    (int x, int y) this[int i] { get => default; set {} }
    void M(object o, bool b)
    {
        b = /*<bind>*/o is C { [1]: (var x, var y) }/*</bind>*/;
    }
}
```

---
In reply to: [254445327](https://github.com/dotnet/roslyn/pull/33178#discussion_r254445327) [](ancestors = 254445327)